### PR TITLE
RFL-1510: Add support for V3 authentication (JWE), scoping attributes.

### PR DESCRIPTION
**Pull request by impl**

### CODE REVIEW

File: src/main/java/io/reflect/token/TokenBuilder.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    /**
+     * Sets the given attribute in this token.
+     *
+     * @param name the name of the attribute
+     * @param value the attribute's value, which must be serializable to JSON
+     * @return this token builder
+     */
+    public TokenBuilder putAttribute(String name, Object value) {
```
Comment (bradhe) [inline src/main/java/io/reflect/token/TokenBuilder.java]: By convention, this should likely be `setAttribute`

**Review by kyleterry — APPROVED** [review]
I wonder if this was a swagger thing

File: src/main/java/io/reflect/token/TokenUtil.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            ByteBuffer secretKeyBytes = ByteBuffer.wrap(new byte[16]);
+            secretKeyBytes.putLong(secretKeyUUID.getMostSignificantBits());
+            secretKeyBytes.putLong(secretKeyUUID.getLeastSignificantBits());
+
+            return new SecretKeySpec(secretKeyBytes.array(), "AES");
+        } catch (IllegalArgumentException iae) {
+            throw new InvalidSecretKeyException();
+        }
+    }
+}
```
Comment (bradhe) [inline src/main/java/io/reflect/token/TokenUtil.java]: Meh, the only utils here are related to secret keys so I'd move this to `SecretKeyUtils`

File: src/main/java/io/reflect/token/Parameter.java [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
-import com.google.gson.GsonBuilder;
-
+import java.io.IOException;
 import java.util.Arrays;
 
-public class Parameter {
+import net.minidev.json.JSONAwareEx;
+import net.minidev.json.JSONObject;
+import net.minidev.json.JSONStreamAwareEx;
+import net.minidev.json.JSONStyle;
```
Comment (bradhe) [inline src/main/java/io/reflect/token/Parameter.java:9]: Just curious--why move away from `gson`?

**Review by bradhe — APPROVED** [review]
Mostly suggestions--feel free to ignore them ;)

File: src/main/java/io/reflect/token/Parameter.java (hunk 3 above)
Comment (impl) [reply src/main/java/io/reflect/token/Parameter.java:9]: This is what the JWT library uses internally, so I didn't really have a choice.

**Pull request merged**